### PR TITLE
List interactive

### PR DESCRIPTION
**Pull request by Jmagero**

### CODE REVIEW

**Comment by Jmagero** [comment]
- [x] Added method related to drag / drop and sorting functionality.
- [x] Added method related to the status updates (completed: true / false)
- [x]  Added event listeners to list items (dragstart, dragover, drop) and to checkbox (change)
- [x] Stored the updated array of items in local storage, so the user gets the correct list values after the page reloads.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hello @Jmagero** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

- here you can find something about the retrive and remove in JS [usage of localstorage in JS](https://www.taniarascia.com/how-to-use-local-storage-with-javascript/) 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Your project looks like :

https://user-images.githubusercontent.com/61562626/126538918-5f063fb6-b1f7-4fca-8de1-6ae0026fd1d2.mp4

## Original : 
todolist




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/localStorage.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const getTasks = () => {
+  if (localStorage.getItem('tasksKey')) {
+    const loadTasks = JSON.parse(localStorage.getItem('tasksKey'));
+    loadTasks.forEach((t) => {
+      tasks.push(t);
+    });
+    addTasks();
+  }
+};
```
Comment (dasileker) [inline src/localStorage.js:17]: - [ ] Please add the added taskskey you have the localStorage but it's not displayed in the Showtask.

File: src/showTasks.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            `;
+    } else {
+      li.innerHTML = `
+              <div>
+              <input class="checkbox m-2" type="checkbox" id="${e.index}">
+              <label for="${e.index}">${e.description}</label>
+              </div>
+              <i class="fas fa-ellipsis-v"></i>
+            `;
+    }
```
Comment (dasileker) [inline src/showTasks.js]: - [ ] Please add the added task from localStorage to the addTask.

**Comment by Jmagero** [comment]
helo Amine

thanks for the review but I didn't understand your comments

**Comment by dasileker** [comment]
> Hello Amine
> 
> thanks for the review but I didn't understand your comments

Hello @Jmagero, the reply is clear, your two functions for the localstorage, and the add is not displaying the add task or the dragging options. 

neither the checkbox, your code looks good but not functional.

**Comment by Jmagero** [comment]
- [x] Made code functional

File: src/index.js [hunk 3]
```diff
@@ -2,34 +2,25 @@ import './style.css';
...
-    this.index = index;
-    this.description = description;
-    this.completed = completed;
-  }
-}
-const task1 = new Task(1, 'first task', false);
-const task2 = new Task(2, 'second task', false);
-const task3 = new Task(3, 'third task', true);
-const task4 = new Task(4, 'forth task', true);
-tasks.push(task1, task2, task3, task4);
```
Comment (Dipeshtwis) [inline src/index.js:18]: - [ ] As you have deleted the default task, I wasn't able to check the drag and drop and completed functionality. Please add some task and also check all the functionality is working properly or not.

Screenshot from 2021-07-22 04-01-19

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Jmagero, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- I would highly recommend to add the `PR description` as well so that reviewer knows what is this `PR` for. :smile: 

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**

-------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Jmagero** [comment]
Helo 
I added a functionality for adding tasks and the tasks show when the user loads

**Comment by Dipeshtwis** [comment]
> Helo
> I added a functionality for adding tasks and the tasks show when the user loads

Hello @Jmagero, 

I tried to create some task but it is not showing in `UI`. But, I can see it's in the `localstorage`. So please fix that so that user can check your project by adding and seeing this in `UI`. Below is the screenshot as well. Also, adding and removing task is in the next milestone, so what I expected is some default task there. Hope it will clear to you. :pray: 

Screenshot from 2021-07-22 04-23-59


Thanks
**_Dipesh Kumar_**

**Comment by Jmagero** [comment]
@ Helo Dipeshtwis 
Thanks for the quick response.
You are not able to see the tasks because I stored the tasks in the localStorage and the user is able to see after the page reloads as this requirement.
**Store the updated array of items in local storage, so the user gets the correct list values after the page reloads**

**Comment by Dipeshtwis** [comment]
> @ Helo Dipeshtwis
> Thanks for the quick response.
> You are not able to see the tasks because I stored the tasks in the localStorage and the user is able to see after the page reloads as this requirement.
> **Store the updated array of items in local storage, so the user gets the correct list values after the page reloads**

Hello @Jmagero,
I can see now. Sorry for that, I don't wanted to mislead you :pray:
I can see that project is working well. Please go ahead and submit for new review, I'm sure you'll get approval this time. :smile: 

Thanks 
**_Dipesh Kumar_**

**Comment by Jmagero** [comment]
@Dipeshtwis 
Is it okay to confirm if I meet all the requirements. This is the second time am repeating this module and if am not approved in my last submission I will be kicked out.
I am sorry I know this is too much to ask.

**Comment by Dipeshtwis** [comment]
> @Dipeshtwis
> Is it okay to confirm if I meet all the requirements. This is the second time am repeating this module and if am not approved in my last submission I will be kicked out.
> I am sorry I know this is too much to ask.

I checked your project and it seems all are working perfect. So, I think you'll surely deserve approval.

**Comment by Jmagero** [comment]
resubmission

**Comment by Davidosky007** [comment]
Hi @Jmagero, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**